akka-stream/src/main/scala/akka/stream/Shape.scala

```diff
@@ -6,10 +6,28 @@ package akka.stream
 import scala.collection.immutable
 import scala.collection.JavaConverters._
 
+/**
+ * An input port of a StreamLayout.Module. This type logically belongs
+ * into the impl package but must live here due to how `sealed` works.
```

Comment:
Alternative is to really move it into impl, to the StreamLayout file, and mark it as INTERNAL API. "sealed" only works in scala anyway.